lxd/instance/instance_utils.go

```diff
@@ -1297,6 +1298,13 @@ func ExpandInstanceConfig(config map[string]string, profiles []api.Profile) map[
 		expandedConfig[k] = v
 	}
 
+	// Apply global config overriding
+	if globalConfig != nil {
```

Comment:
Given that global config is hierarchically speaking, applied first, and then the profiles and local config are applied afterwards. It would make more sense to me for this to be moved before the `// Apply all the profiles` line and just do:

```go
// Apply global config.
if globalConfig != nil {
    expandedConfig["migration.stateful"] = strconv.FormatBool(globalConfig.InstancesMigrationStateful())
}
```

Also i was thinking because the `globalConfig` concept doesn't differentiate between the config item being unset and being explicitly set to the default value, this means that the value will always appear to be set to something in `lxc config show --expanded` now which seems at odds with how profiles and local config work.

So how about this instead so it only shows if its set to true:

```go
// Apply global config.
if globalConfig != nil {
    globalInstancesMigrationStateful := strconv.FormatBool(globalConfig.InstancesMigrationStateful()
    if globalInstancesMigrationStateful {
        expandedConfig["migration.stateful"] = strconv.FormatBool(globalInstancesMigrationStateful)
    }
}
```

Reply:
fixed